ethereum_history_api/circuits/lib/src/verifiers/transaction/rlp.nr

```diff
@@ -55,7 +56,7 @@ pub(crate) fn assert_tx_rlp_equals<MAX_DATA_LEN>(
 
     fragments[to_idx + 2].assert_string_eq("Data", tx_rlp, tx.data);
 
-    fragments[signature_idx].assert_string_eq("V", tx_rlp, Into::into(byte_value([tx.v as u8])));
-    fragments[signature_idx + 1].assert_string_eq("R", tx_rlp, Into::into((tx.r, HASH_LENGTH)));
-    fragments[signature_idx + 2].assert_string_eq("S", tx_rlp, Into::into((tx.s, HASH_LENGTH)));
+    fragments[signature_idx].assert_string_eq("V", tx_rlp, byte_value([tx.v as u8]));
+    fragments[signature_idx + 1].assert_string_eq("R", tx_rlp, BoundedVec { storage: tx.r, len: HASH_LENGTH });
+    fragments[signature_idx + 2].assert_string_eq("S", tx_rlp, BoundedVec { storage: tx.s, len: HASH_LENGTH });
```

Comment:
Should we introduce syntax sugar `array_to_bounded_vec`, so that we can have:
```rust
 fragments[signature_idx + 2].assert_string_eq("S", tx_rlp, array_to_bounded_vec(tx.s));
 ```

Reply:
Ideally we could introduce `From` trait, but due to orphan rule we can't.
So we could introduce `As` trait:
```rust
trait As<T> {
  fn as(self) -> T {
  }
}

impl As<BoundedVec> for Array {

}

fragments[signature_idx + 2].assert_string_eq("S", tx_rlp, tx.s.as());
```

And do a PR to Noir stlib to introduce From/Into.

Reply:
#next PR

Reply:
Another syntax to convert array to `BoundedVec` is
```rust
BoundedVec::new().extend_from_array(array);
```